src/main/java/org/primefaces/component/api/UICalendar.java

```diff
@@ -217,6 +249,7 @@ public void setLabelledBy(String labelledBy) {
     public Converter getConverter() {
         Converter converter = super.getConverter();
 
+        //TODO: Why does it matter whether Client-Side-Validation is enabled?
```

Comment:
no idea, could you try to debug and/or remove this and check what happens?

Reply:
I tried both variant´s (via primefaces.CLIENT_SIDE_VALIDATION) but got no real idea. Thought about throwing this out.
Maybe there is some missalignement between the client-side-validation and the server-side-validation. And this code tries to work arround this issue. Hoped someone reminds why this was implemented this way some time ago.

Reply:
Checked how this was implemented in 6.2. (Before Datepicker)
https://github.com/primefaces/primefaces/blob/6_2/src/main/java/org/primefaces/component/calendar/CalendarRenderer.java
6.2 did not had this (from my perspective) awkward behaviour depending on Client-Side-Validation.
So i´ll throw this out. At least as long until someone come´s up with an argument, why this needs to be this way.